### PR TITLE
Add various project URLs for PyPI

### PR DESCRIPTION
**Pull request by mkoeppe**
The changed file, `m4/pyproject_toml_metadata.m4`, controls the metadata of the modularized distributions.

The same changes should be done to **sagemath-standard** and **sage-conf** after #37902, #36561 are merged.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: m4/pyproject_toml_metadata.m4 [hunk 1]
```diff
@@ -17,5 +17,10 @@ classifiers = [
     "Programming Language :: Python :: Implementation :: CPython",
     "Topic :: Scientific/Engineering :: Mathematics",
 ]
-urls = {Homepage = "https://www.sagemath.org"}
+urls = {download = "https://doc.sagemath.org/html/en/installation/index.html",
+        "release notes" = "https://github.com/sagemath/sage/releases",
```
Comment (kwankyu) [inline m4/pyproject_toml_metadata.m4:21]: Is "download" correct?

Comment (kwankyu) [reply m4/pyproject_toml_metadata.m4:21]: The url names are capitalized?

https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls

Comment (mkoeppe) [reply m4/pyproject_toml_metadata.m4:21]: I've been following https://github.com/scipy/scipy/blob/main/pyproject.toml#L116 and some other projects. When displaying this info, PyPI automatically sentence-cases the URL names.

Comment (kwankyu) [reply m4/pyproject_toml_metadata.m4:21]: I see.

"download" points to the installation guide. How about "installation"?

Comment (mkoeppe) [reply m4/pyproject_toml_metadata.m4:21]: > "download" points to the installation guide.

Yes, this is deliberate; same as on our website

Comment (kwankyu) [reply m4/pyproject_toml_metadata.m4:21]: Hmm. OK.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks!

**Comment by antonio-rojas** [comment]
This breaks build of optional distributions. For `sagemath-bliss` I get
```
python_packages = []
python_modules = ['sage.all__sagemath_bliss', 'sage.graphs.all__sagemath_bliss']
cython_modules = [<setuptools.extension.Extension('sage.graphs.bliss') at 0x78611b76d400>]
************************************************************************
Traceback (most recent call last):
  File "/build/sagemath-git/src/sage/pkgs/sagemath-bliss/setup.py", line 60, in <module>
    setup(
  File "/usr/lib/python3.12/site-packages/setuptools/__init__.py", line 104, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 158, in setup
    dist.parse_config_files()
  File "/usr/lib/python3.12/site-packages/setuptools/dist.py", line 631, in parse_config_files
    pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
  File "/usr/lib/python3.12/site-packages/setuptools/config/pyprojecttoml.py", line 74, in apply_configuration
    config = read_configuration(filepath, True, ignore_option_errors, dist)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/config/pyprojecttoml.py", line 110, in read_configuration
    asdict = load_file(filepath) or {}
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/config/pyprojecttoml.py", line 36, in load_file
    return tomllib.load(file)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 66, in load
    return loads(s, parse_float=parse_float)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 102, in loads
    pos = key_value_rule(src, pos, out, header, parse_float)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 326, in key_value_rule
    pos, key, value = parse_key_value_pair(src, pos, parse_float)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 369, in parse_key_value_pair
    pos, value = parse_value(src, pos, parse_float)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 620, in parse_value
    return parse_inline_table(src, pos, parse_float)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 445, in parse_inline_table
    pos, key, value = parse_key_value_pair(src, pos, parse_float)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 360, in parse_key_value_pair
    pos, key = parse_key(src, pos)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 374, in parse_key
    pos, key_part = parse_key_part(src, pos)
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tomllib/_parser.py", line 404, in parse_key_part
    raise suffixed_err(src, pos, "Invalid initial character for a key part")
tomllib.TOMLDecodeError: Invalid initial character for a key part (at line 39, column 79)
************************************************************************
```
The relevant `pyproject.toml` is
```
# Generated by SAGE_ROOT/bootstrap; do not edit
[build-system]
# Minimum requirements for the build system to execute.
requires = [
    'setuptools >= 68.1.1',
    'sage-conf ~= 10.5b3',
    'sage-setup ~= 10.5b3',
    'sagemath-environment ~= 10.5b3',
    'cython >=3.0, != 3.0.3, <4.0',
    'cysignals >=1.11.4',
    'pkgconfig',
]
build-backend = "setuptools.build_meta"

[project]
name = "sagemath-bliss"
description = "Sage: Open Source Mathematics Software: Graph (iso/auto)morphisms with bliss"
dependencies = [
    'cysignals >=1.11.4',
]
dynamic = ["version"]
license = {text = "GNU General Public License (GPL) v2 or later"}
authors = [{name = "The Sage Developers", email = "sage-support@googlegroups.com"}]
classifiers = [
    "Development Status :: 6 - Mature",
    "Intended Audience :: Education",
    "Intended Audience :: Science/Research",
    "License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)",
    "Operating System :: POSIX",
    "Operating System :: MacOS :: MacOS X",
    "Programming Language :: Python :: 3 :: Only",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
    "Programming Language :: Python :: Implementation :: CPython",
    "Topic :: Scientific/Engineering :: Mathematics",
]
urls = {download = "https://doc.sagemath.org/html/en/installation/index.html",
        "release notes" = "https://github.com/sagemath/sage/releases",
        source = "https://github.com/sagemath/sage",
        documentation = "https://doc.sagemath.org",
        homepage = "https://www.sagemath.org",
        tracker = "https://github.com/sagemath/sage/issues"}
requires-python = ">=3.9, <3.13"

[project.readme]
file = "README.rst"
content-type = "text/x-rst"

[project.optional-dependencies]
test = ['sagemath-repl ~= 10.5b3',]

[tool.setuptools]
include-package-data = false

[tool.setuptools.dynamic]
version = {file = ["VERSION.txt"]}
```

**Comment by antonio-rojas** [comment]
According to the TOML spec, newlines are not allowed inside inline tables

https://toml.io/en/v1.0.0#inline-table

**Pull request merged**